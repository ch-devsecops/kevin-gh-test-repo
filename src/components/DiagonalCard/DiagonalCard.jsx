import React from 'react';
import PropTypes from 'prop-types';
import { Box } from '@honda-canada/design-system-react';
import themeStyles from './DiagonalCard.styles';

const Container = themeStyles.apply(Box, 'Container');
const ContentBox = themeStyles.apply(Box, 'ContentBox');
const ImageBox = themeStyles.apply(Box, 'ImageBox');
const ImageWrapper = themeStyles.apply(Box, 'ImageWrapper');
const TransitionImageWrapper = themeStyles.apply(Box, 'TransitionImageWrapper');
const FadeUp = themeStyles.apply(Box, 'FadeUp');

const DiagonalCardBase = ({
  title,
  bodyText,
  image,
  cta,
  transitionImage,
  isReversing,
  isActive = true,
  inView,
  gtmTags = {},
  isFullHeight,
  mb,
  isDark,
}) => {
  const isOpen = inView && isActive;

  return (
    <Container
      data-gtm-category={gtmTags.category}
      data-gtm-component-type={gtmTags.type}
      isFullHeight={isFullHeight}
      mb={mb}
      backgroundColor={isDark ? 'black' : 'transparent'}
    >
      <ContentBox>
        <FadeUp isOpen={isOpen} delay="0.3s" mb="xs">
          {title}
        </FadeUp>

        <FadeUp isOpen={isOpen} delay="0.4s" mb={['m', 'm']} id="diagonal-card-body">
          {bodyText}
        </FadeUp>

        <FadeUp isOpen={isOpen} delay="0.5s">
          {cta}
        </FadeUp>
      </ContentBox>

      <ImageBox>
        {transitionImage && (
          <TransitionImageWrapper isReversing={isReversing} isOpen={isOpen}>
            {transitionImage}
          </TransitionImageWrapper>
        )}
        <ImageWrapper isOpen={isOpen} isReversing={isReversing}>
          {image}
        </ImageWrapper>
      </ImageBox>
    </Container>
  );
};

DiagonalCardBase.propTypes = {
  title: PropTypes.node,
  bodyText: PropTypes.node,
  image: PropTypes.element,
  cta: PropTypes.element,
  /**
   * In a series of cards, the image for the next card. This image
   * is shown whem transitioning between this card and another one.
   */
  transitionImage: PropTypes.element,
  /**
   * In a series of cards, is this the active Card?
   */
  isActive: PropTypes.bool,
  /**
   * Should the card transition in reverse?
   */
  isReversing: PropTypes.bool,
  inView: PropTypes.bool,
  gtmTags: PropTypes.shape({
    type: PropTypes.string,
    category: PropTypes.string,
  }),
  isFullHeight: PropTypes.bool,
  mb: PropTypes.string,
  isDark: PropTypes.bool,
};

export default DiagonalCardBase;
