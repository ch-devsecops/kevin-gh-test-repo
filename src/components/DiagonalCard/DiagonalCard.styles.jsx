import css from '@styled-system/css';
import { ThemeStyles } from '@honda-canada/design-system-react';

const themeStyles = new ThemeStyles();

themeStyles.addBaseStyles('Container')(({ isFullHeight }) =>
  css({
    display: 'flex',
    flexDirection: ['column-reverse', 'column'],
    justifyContent: ['flex-end', 'flex-start'],
    pt: '50px',
    height: isFullHeight ? 'calc(100vh - 50px)' : '100%',
  }),
);

themeStyles.addBaseStyles('ContentBox')(
  css({
    maxWidth: ['100%', '636px'],
    width: '100%',
    paddingX: [20, 0],
    mt: ['l', 0],
    ml: [0, 'big'],
  }),
);

themeStyles.addBaseStyles('ImageBox')(
  css({
    position: 'relative',
    marginTop: [0, '-5%'],
    paddingTop: '56.25%',
    pointerEvents: 'none',
  }),
);

themeStyles.addBaseStyles('ImageWrapper')(({ isOpen, isReversing }) =>
  css({
    position: 'absolute',
    top: 0,
    height: '100%',
    left: isOpen ? 0 : '50%',
    width: isOpen ? '100%' : 0,
    transition: 'left ease-out, width ease-out',
    transitionDuration: isReversing ? 0 : 't1',
  }),
);

themeStyles.addBaseStyles('TransitionImageWrapper')(({ isReversing, isOpen }) =>
  css({
    position: 'absolute',
    top: 0,
    height: '100%',
    zIndex: isReversing && isOpen ? 1 : 0,
    left: isReversing && isOpen ? '50%' : 0,
    width: isReversing && isOpen ? 0 : '100%',
    transition: 'left ease-out, width ease-out',
    transitionDuration: 't1',
  }),
);

themeStyles.addBaseStyles('FadeUp')(({ isOpen, delay = 0 }) =>
  css({
    opacity: isOpen ? 1 : 0,
    transform: isOpen ? 'none' : 'translateY(50px)',
    transition: 'opacity ease-out, transform ease-out',
    transitionDuration: 't3',
    transitionDelay: delay,
  }),
);

export default themeStyles;
