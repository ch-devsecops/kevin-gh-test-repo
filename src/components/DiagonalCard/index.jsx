import React, { useContext } from 'react';
import { compiler } from 'markdown-to-jsx';
import { H3, Markdown } from '@honda-canada/design-system-react';
import { InView } from 'react-intersection-observer';
import CTA from '../CTA';
import { mapFieldToDesignSystemImage, mapGTMCategory } from '../../utils/sitecoreFields';
import { stripMarkdownHeading, getTitleComponent } from '../../utils/markdown';
import { ModelExplorationContext } from '../ModelExplorationContext';
import DiagonalCardBase from './DiagonalCard';

const DiagonalCardWrapper = ({ fields, rendering, params }) => {
  const { title, bodyText, ctaType, ctaLink, ctaIcon, mediaImage, gtmCategory, gtmTitle } = fields;

  const modelExplorationContext = useContext(ModelExplorationContext) || {};
  const { isDark } = modelExplorationContext;

  const gtmTags = {
    category: mapGTMCategory(gtmCategory),
    type: rendering?.componentName,
  };

  const cta = ctaType?.value ? (
    <CTA linkField={ctaLink} typeField={ctaType} iconField={ctaIcon} data-gtm-title={gtmTitle?.title} />
  ) : null;

  const image = mapFieldToDesignSystemImage(mediaImage);
  const Title = getTitleComponent(title.value, H3);
  const TitleComponent = (
    <Title color={isDark ? 'white' : 'typographyDefault'}>{compiler(stripMarkdownHeading(title.value))}</Title>
  );
  const BodyText = <Markdown color={isDark ? 'white' : 'typographyDefault'}>{bodyText.value}</Markdown>;

  return (
    <InView>
      {({ inView, ref }) => (
        <div ref={ref}>
          <DiagonalCardBase
            isDark={isDark}
            title={TitleComponent}
            bodyText={BodyText}
            inView={inView}
            image={image}
            cta={cta}
            gtmTags={gtmTags}
            mb={`${params?.bottomMargin || 0}px`}
          />
        </div>
      )}
    </InView>
  );
};

export default DiagonalCardWrapper;
