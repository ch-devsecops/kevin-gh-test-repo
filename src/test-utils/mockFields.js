/* eslint-disable max-len */
/* eslint-disable max-classes-per-file */

// FIXME: jsdoc popup not working on test files
/** Class representing a mock data. */
class MockData {
  /**
   * Add a field.
   * @param {string} key - key of the field obj: i.e. 'title'
   * @param {Object} value - value of the field obj: i.e. { value: 'some text' }.
   */
  add(key, value) {
    this[key] = value;
  }
}

/**
 * Class representing mock fields object.
 * @extends MockData
 *
 * available method: add()
 *
 * default fields: [anchorId, title, bodyText, image, ctaLink, ctaType, ctaIcon, gtmCategory, gtmTitle, contentTitle, subtitle]
 *
 * note: we can add more commonly used fields in the class declaration if needed
 */
class MockFields extends MockData {
  constructor() {
    super();

    this.title = { value: 'Title' };
    this.bodyText = { value: 'BodyText' };
    this.image = {
      value: { src: 'mock.png', alt: 'mock alt' },
    };
    this.ctaLink = {
      value: {
        href: 'www.source.ca',
        title: 'cta aria label',
        text: 'CTA Text',
      },
    };
    this.ctaType = {
      value: 'Tertiary',
    };
    this.ctaIcon = {
      fields: {
        value: {
          value: 'arrowRight',
        },
      },
    };
    this.gtmCategory = {
      fields: {
        value: {
          value: 'Others',
        },
      },
    };
    this.gtmTitle = {
      value: 'gtm title',
    };
    this.contentTitle = {
      value: 'Content Title',
    };
    this.subtitle = {
      value: 'Content Subtitle',
    };
    this.seoH1 = {
      value: '',
    };
    this.anchorId = {
      value: 'anchor-id',
    };
    this.ctaLabel = {
      value: 'CTA Label',
    };
  }
}

/**
 * Class representing a mock rendering object
 * @extends MockData
 *
 * available method: add()
 */
class MockRendering extends MockData {
  /**
   * Create a mock rendering object
   * @param {string} componentName - adds componentName
   */
  constructor(componentName) {
    super();

    this.componentName = componentName;
  }
}

export { MockFields, MockRendering };
