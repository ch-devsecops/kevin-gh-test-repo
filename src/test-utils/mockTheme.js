/**
 * Mock styled-components theme, if you want to test how your component
 * implements theme styles. Import as-is, or spread in any
 * additional values you need.
 *
 * Example usage:

  it('should apply the bgColor param', () => {
    const { colors } = MockTheme;
    const { container } = render(
      <MyComponent
        fields={mockFields}
        params={{ bgColor: 'Gray' }}
      />,
    );
    const wrapper = container.querySelector('section');

    expect(wrapper).toHaveStyleRule('background-color', colors.grey[5]);
  });
 */

import { HONDA_THEME_NAME } from '../utils/constants';

const mockTheme = {
  letterSpacings: {
    sm: '0.5px',
    md: '1px',
  },
  lineHeights: {
    sm: '22px',
    md: '26px',
  },
  colors: {
    white: '#fff',
    transparent: 'transparent',
    grey: ['#707070', '#C7C7C7', '#D5D5D5', '#EAEAEA', '#F8F8F8', '#F5F5F5'],
    black: '#2C2C2C',
    red: '#DA2C2E',
    redShades: ['rgba(200, 16, 46, 0.1)'],
    aodaRed: '#D92629',
    lightRed: '#EC9597',
    darkRed: '#A21C1D',
    blue: '#334C96',
    darkBlue: '#002D62',
    lightBlue: '#00447826',
    yellow: ['#F9CA52', '#EDAB00', '#8D763F'],
    fadedBlue: '#ADB7D5',
    primary: '#DA2C2E',
    primaryAoda: '#D92629',
    primaryDark: '#A21C1D',
    primaryLight: '#EC9597',
    secondary: '#2C2C2C',
    secondaryLight: '#C7C7C7',
    tertiary: '#002D62',
    tertiaryLight: '#334C96',
    tertiaryDisabled: '#ADB7D5',
    danger: '#DA2C2E',
    warning: '#2C2C2C',
    warningLight: '#EAEAEA',
    success: '#334C96',
    successLight: '#d9e0e7',
    disabled: '#C7C7C7',
    typographyDefault: '#2C2C2C',
  },
  space: [0, 4, 8, 12, 16, 24, 32, 48, 64, 96, 128, 144, 192, 256],
  borders: [0, '1px solid', '2px solid', '3px solid'],
  breakpoints: ['768px', '992px', '1200px'],
  fontWeights: {
    regular: 'normal',
    bold: 700,
  },
  fontSizes: {
    sm: '12px',
  },
  shadows: {
    smooth: {
      md: '0.1px 0.3px 0.3px rgba(0, 0, 0, 0.02),0.3px 0.7px 0.9px rgba(0, 0, 0, 0.028),0.7px 1.4px 1.8px rgba(0, 0, 0, 0.035),1.5px 2.9px 3.7px rgba(0, 0, 0, 0.047),4px 8px 10px rgba(0, 0, 0, 0.07)',
    },
    box: {
      subNav: '0px 2px 4px rgba(0, 0, 0, 0.1)',
    },
  },
  zIndices: {
    accordionItemTitle: 200,
    modal: 13001,
    sectionNav: 13000,
  },
  transitionDuration: {
    t1: '1s',
  },
  transitionTimingFunction: {
    default: 'ease-in-out',
    out: 'ease-out',
    in: 'ease-in',
  },
  name: HONDA_THEME_NAME,
  fonts: {
    default: 'Helvetica',
    bold: 'Cooper Black',
    heading: 'Roboto',
  },
  header: {
    desktop: {},
    mobile: {},
  },
  icons: {},
  logIconWarnings: false,
};

export default mockTheme;
