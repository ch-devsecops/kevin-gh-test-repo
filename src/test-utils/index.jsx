import React from 'react';
import { render } from '@testing-library/react';
import { ThemeProvider } from 'styled-components';
import { SitecoreContext } from '@sitecore-jss/sitecore-jss-react';
import { BrowserRouter } from 'react-router-dom';
import { OptionalVideoModalContextProvider } from '@honda-canada/design-system-react';
import UserLocationProvider from '@honda-canada/user-location';
import { configureAxe } from 'jest-axe';
import ModelExplorationContextProvider from '../components/ModelExplorationContext';
import ModelFiltersContextProvider from '../components/ModelFiltersContext';
import { MockFields, MockRendering } from './mockFields';
import MockTheme from './mockTheme';
import 'intersection-observer';
import { ACURA_THEME_NAME } from '../utils/constants';

const generateJssWrapper = context =>
  // eslint-disable-next-line react/display-name
  function ({ children }) {
    return (
      <SitecoreContext
        context={{
          language: 'en',
          site: { name: ACURA_THEME_NAME },
          ...context?.sitecoreContext,
        }}
        componentFactory={{}}
      >
        <BrowserRouter>
          <UserLocationProvider>
            <ThemeProvider theme={context?.theme || MockTheme}>
              <OptionalVideoModalContextProvider>
                <ModelExplorationContextProvider>
                  <ModelFiltersContextProvider>{children}</ModelFiltersContextProvider>
                </ModelExplorationContextProvider>
              </OptionalVideoModalContextProvider>
            </ThemeProvider>
          </UserLocationProvider>
        </BrowserRouter>
      </SitecoreContext>
    );
  };
/**
 * A custom implementation of the testing-library `render` function that
 * wraps tests in an typical JSS app's context providers.
 * Note that you can include a `context` object to set or override values
 * for SitecoreContext and ThemeContext.
 * @param {ReactElement} component - the component you want testing-library to render
 * @param {object} [options] - see https://testing-library.com/docs/react-testing-library/api/#render-options
 * @param {string} [options.context] - default values for SitecoreContext, ThemeProvider
 * @param {string} [options.context.sitecoreContext] - value for SitecoreContext
 * @param {string} [options.context.theme] - value for ThemeProvider
 * @example
 * const { getByText } = jssRender(
 *  <MyComponent />, { context: { sitecoreContext: { language: 'fr' } } }
 * );
 * @returns {function(): void} custom testing-library render function
 */
const jssRender = (component, options = {}) => {
  const { context } = options;
  // eslint-disable-next-line no-param-reassign
  delete options.context;

  return render(component, {
    wrapper: generateJssWrapper(context),
    ...options,
  });
};

const generateWrapper = context =>
  // eslint-disable-next-line react/display-name
  function ({ children }) {
    return (
      <BrowserRouter>
        <ThemeProvider theme={context?.theme || MockTheme}>
          <OptionalVideoModalContextProvider>{children}</OptionalVideoModalContextProvider>
        </ThemeProvider>
      </BrowserRouter>
    );
  };

/**
 * A custom implementation of the testing-library `render` function that
 * wraps tests in typical design-system context providers.
 * Note that you can include a `theme` object to set or override values
 * for ThemeContext.
 * @param {ReactElement} component - the component you want testing-library to render
 * @param {object} [options] - see https://testing-library.com/docs/react-testing-library/api/#render-options
 * @param {string} [options.theme] - default value for ThemeProvider
 * @example
 * const { getByText } = render(
 *  <MyComponent />, { theme: { space: { wrapperGap: '15px' } } }
 * );
 * @returns {function(): void} custom testing-library render function
 */
const customRender = (component, options = {}) => {
  const { theme } = options;
  // eslint-disable-next-line no-param-reassign
  delete options.theme;

  return render(component, {
    wrapper: generateWrapper(theme),
    ...options,
  });
};

// disable accessibility checks for errors that are usually trigged by content mistakes
const axe = configureAxe({
  rules: {
    'duplicate-id': { enabled: false },
    'heading-order': { enabled: false },
  },
});

export * from 'jest-axe';
export { axe };
export * from '@testing-library/react';
export { customRender as render, jssRender };
export { MockFields, MockRendering, MockTheme };
