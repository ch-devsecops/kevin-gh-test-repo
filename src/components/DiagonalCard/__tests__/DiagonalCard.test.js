import 'jest-styled-components';
import React from 'react';
import { jssRender, cleanup, MockRendering, MockFields, axe } from 'test-utils';
import DiagonalCard from '../DiagonalCard';

afterEach(cleanup);

const mockFields = new MockFields();
const mockRendering = new MockRendering('DiagonalCard');

describe('<DiagonalCard>', () => {
  const renderMock = () =>
    jssRender(
      <DiagonalCard fields={mockFields} rendering={mockRendering} title="mock-title" bodyText="mock-body-text" />,
    );

  it('should have no accessibility violations', async () => {
    const { container } = renderMock();

    expect(await axe(container)).toHaveNoViolations();
  });

  it('should match the snapshot', async () => {
    const { container } = renderMock();

    expect(container).toMatchSnapshot();
  });

  it('should render provided body text', () => {
    const { container } = renderMock();
    const body = container.querySelector('div:nth-child(2)');

    expect(body).toHaveTextContent('mock-body-text');
  });
});
