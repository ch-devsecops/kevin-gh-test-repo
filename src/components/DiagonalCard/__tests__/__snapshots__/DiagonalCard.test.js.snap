// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DiagonalCard> should match the snapshot 1`] = `
.c0 {
  background-color: transparent;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column-reverse;
  -ms-flex-direction: column-reverse;
  flex-direction: column-reverse;
  -webkit-box-pack: end;
  -webkit-justify-content: flex-end;
  -ms-flex-pack: end;
  justify-content: flex-end;
  padding-top: 50px;
  height: 100%;
}

.c1 {
  max-width: 100%;
  width: 100%;
  padding-left: 20px;
  padding-right: 20px;
  margin-top: l;
  margin-left: 0;
}

.c2 {
  margin-bottom: xs;
  opacity: 0;
  -webkit-transform: translateY(50px);
  -ms-transform: translateY(50px);
  transform: translateY(50px);
  -webkit-transition: opacity ease-out,-webkit-transform ease-out;
  -webkit-transition: opacity ease-out,transform ease-out;
  transition: opacity ease-out,transform ease-out;
  -webkit-transition-duration: t3;
  transition-duration: t3;
  -webkit-transition-delay: 0.3s;
  transition-delay: 0.3s;
}

.c3 {
  margin-bottom: m;
  opacity: 0;
  -webkit-transform: translateY(50px);
  -ms-transform: translateY(50px);
  transform: translateY(50px);
  -webkit-transition: opacity ease-out,-webkit-transform ease-out;
  -webkit-transition: opacity ease-out,transform ease-out;
  transition: opacity ease-out,transform ease-out;
  -webkit-transition-duration: t3;
  transition-duration: t3;
  -webkit-transition-delay: 0.4s;
  transition-delay: 0.4s;
}

.c4 {
  opacity: 0;
  -webkit-transform: translateY(50px);
  -ms-transform: translateY(50px);
  transform: translateY(50px);
  -webkit-transition: opacity ease-out,-webkit-transform ease-out;
  -webkit-transition: opacity ease-out,transform ease-out;
  transition: opacity ease-out,transform ease-out;
  -webkit-transition-duration: t3;
  transition-duration: t3;
  -webkit-transition-delay: 0.5s;
  transition-delay: 0.5s;
}

.c5 {
  position: relative;
  margin-top: 0;
  padding-top: 56.25%;
  pointer-events: none;
}

.c6 {
  position: absolute;
  top: 0;
  height: 100%;
  left: 50%;
  width: 0;
  -webkit-transition: left ease-out,width ease-out;
  transition: left ease-out,width ease-out;
  -webkit-transition-duration: 1s;
  transition-duration: 1s;
}

@media screen and (min-width:768px) {
  .c0 {
    -webkit-flex-direction: column;
    -ms-flex-direction: column;
    flex-direction: column;
    -webkit-box-pack: start;
    -webkit-justify-content: flex-start;
    -ms-flex-pack: start;
    justify-content: flex-start;
  }
}

@media screen and (min-width:768px) {
  .c1 {
    max-width: 636px;
    padding-left: 0;
    padding-right: 0;
    margin-top: 0;
    margin-left: big;
  }
}

@media screen and (min-width:768px) {
  .c3 {
    margin-bottom: m;
  }
}

@media screen and (min-width:768px) {
  .c5 {
    margin-top: -5%;
  }
}

<div>
  <div
    class="c0"
  >
    <div
      class="c1"
    >
      <div
        class="c2"
      >
        mock-title
      </div>
      <div
        class="c3"
        id="diagonal-card-body"
      >
        mock-body-text
      </div>
      <div
        class="c4"
      />
    </div>
    <div
      class="c5"
    >
      <div
        class="c6"
      />
    </div>
  </div>
</div>
`;
